evals/solvers/providers/google/gemini_solver_wav.py

```diff
@@ -0,0 +1,239 @@
+import base64
+import copy
+import io
+import os
+from dataclasses import asdict, dataclass
+import re
+from typing import Any, Dict, Union
+
+import google.api_core.exceptions
+import google.generativeai as genai
+from google.generativeai.client import get_default_generative_client
+import librosa
+import numpy as np
+
+from evals.record import record_sampling
+from evals.solvers.solver import Solver, SolverResult
+from evals.task_state import Message, TaskState
+from evals.utils.api_utils import create_retrying
+
+# Load API key from environment variable
+API_KEY = os.environ.get("GEMINI_API_KEY")
+genai.configure(api_key=API_KEY)
+
+SAFETY_SETTINGS = [
+    {
+        "category": "HARM_CATEGORY_HARASSMENT",
+        "threshold": "BLOCK_NONE",
+    },
+    {
+        "category": "HARM_CATEGORY_HATE_SPEECH",
+        "threshold": "BLOCK_NONE",
+    },
+    {
+        "category": "HARM_CATEGORY_SEXUALLY_EXPLICIT",
+        "threshold": "BLOCK_NONE",
+    },
+    {
+        "category": "HARM_CATEGORY_DANGEROUS_CONTENT",
+        "threshold": "BLOCK_NONE",
+    },
+]
+GEMINI_RETRY_EXCEPTIONS = (
+    google.api_core.exceptions.RetryError,
+    google.api_core.exceptions.TooManyRequests,
+    google.api_core.exceptions.ResourceExhausted,
+)
+
+def _data_url_to_wav(url):
+    if not url.startswith("data:"):
+        raise ValueError("Not a data URL")
+    _, raw_data = url.split(",", 1)
+    return base64.b64decode(raw_data)
+
+
+# TODO: Could we just use google's own types?
+# e.g. google.generativeai.types.content_types.ContentType
+@dataclass
+class GoogleMessage:
```

Comment:
Majority of code in this class is copied from `gemini_solver.py`, is it possible.e to reuse the existing code and subclass if necessary, for example `GeminiSolverWav`?